packages/base/src/util/dragAndDrop/handleDragOver.ts

```diff
@@ -0,0 +1,61 @@
+import type UI5Element from "../../UI5Element.js";
+import type MovePlacement from "../../types/MovePlacement.js";
+import DragRegistry from "./DragRegistry.js";
+
+type DragOverResult = {
+	targetReference: HTMLElement | null;
+	placement: any;
+}
+
+type DragPosition = {
+	element: HTMLElement;
+	placements: MovePlacement[];
+};
+
+/**
+ * Handles the dragover event.
+ */
+function handleDragOver(e: DragEvent, component: UI5Element, position: DragPosition, target: HTMLElement): DragOverResult {
+	const draggedElement = DragRegistry.getDraggedElement();
+	const dragOverResult: DragOverResult = {
+		targetReference: null,
+		placement: null,
+	};
+
+	if (!draggedElement) {
+		return dragOverResult;
+	}
+
+	const placements = position.placements;
+	dragOverResult.targetReference = e.target as HTMLElement;
+
+	const placementAccepted = placements.some(placement => {
+		const beforeItemMovePrevented = !component.fireDecoratorEvent("move-over", {
+			originalEvent: e,
```

Comment:
This originalEvent is only a parameter for the List component, so I'm not sure whether the table and other components should also have it.
Could introduce a flag, but I feel like it would be better to completely remove this param

Reply:
a developer might want to change the behavior when for example control key is pressed while dragging, or might need to access dataTransfer object such information only available in the original event.

Reply:
a developer might want to change the behavior when for example control key is pressed while dragging, or might need to access dataTransfer object that is set at the dragStart such information only available in the original event. IMHO originalEvent is mandatory

Reply:
That does make sense. Question now is, whether components such as Tree and TabContainer pass the original event as well (which is currently not the case)
Maybe @dimovpetar can weigh in on that

Reply:
It is true that the `List` component has this parameter. It is needed to enable the `TabContainer` to implement specific tree-like behavior for its inner list. This can be considered as workaround for the TabContainer, which we can hopefully deprecate and remove in 3.0, once the design of the TabContainer is updated to replace the inner list with a tree.
Whether other components need such parameter or not has not been discussed. Since there is no request for it, I think you can omit it.

Reply:
I am insist to keep it. it does not hurt but also allows many use-case which would not be possible without original event.